File: go/test/protocol/test_ack_kind.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

import (
	"errors"

	"gopkg.in/yaml.v3"
)

type TestAckKind int

const (
	Success TestAckKind = iota
	Fail
	Drop
)

func (k TestAckKind) String() string {
	return [...]string{"Success", "Fail", "Drop"}[k]
}

func (k *TestAckKind) UnmarshalYAML(value *yaml.Node) error {
	if value.Kind != yaml.ScalarNode {
		return errors.New("TestAckKind must be ScalarNode")
	}

	switch value.Value {
	default:
		return errors.New("unrecognized TestAckKind")
	case "success":
		*k = Success
		return nil
	case "fail":
		*k = Fail
		return nil
	case "drop":
		*k = Drop
		return nil
	}
}

File: rust/azure_iot_operations_protocol/tests/metl/test_ack_kind.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use serde::Deserialize;

#[derive(Clone, Deserialize, Debug, PartialEq)]
pub enum TestAckKind {
    #[serde(rename = "success")]
    Success,

    #[serde(rename = "fail")]
    Fail,

    #[serde(rename = "drop")]
    Drop,
}
